### PR TITLE
Add tests on SQL Server back ends

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
With MSSQL support (SQL Server) around the corner for SCDB (https://github.com/ssi-dk/SCDB/pull/77),
`diseasystore` should be made ready to run its tests on MSSQL.

This PR adapts our testing suite to the new one from AEF-DDF, but does not address all of the issues on all backends. 

### Approach
* We now use the new workflows on AEF-DDF (which are updated to configure MSSQL)

* `helper-setup.R` updated with the new version on AEF-DDF (to pick up the testing connections)

I also needed to manually convert some `DBI::Id`s to `character` in this migration.

`pmax` and `pmin` did not exist on SQL Server, so I have replace these with `ìfelse` which works in our case.

With `SCDB` v0.4 around the corner, I have adapted the CI tests to run with both v0.3 and v0.4.
We aim for the no issues on the v0.4 branch but allow issues for v0.3 as long as it does not interfere with the simplest case (SQLite without schema).
This PR will not fix all of these issues, but adding the workflows allows us to track our progress with fixing the issues.

### Checklist

* [ ] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [x] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

**Comment by RasmusSkytte** [comment]
I am converting this to a draft while waiting for these PRs to be merged:

* [ ] https://github.com/ssi-dk/AEF-DDF/pull/7
* [ ] https://github.com/ssi-dk/SCDB/pull/80
* [ ] https://github.com/ssi-dk/SCDB/issues/79

Once these are merged, we get can do similar updates to this PR to fully test the (correctly configured) workflows.
This should give us the best possible basis for testing SQL Server compatability

**Comment by RasmusSkytte** [comment]
Tests are failing on SQLite with schema, but as I write in the PR description, we only add the testing pipeline here.
I will likely need a version 0.4 of SCDB to nicely fix these tests.

**Comment by RasmusSkytte** [comment]
It was possible to fix the SQLite w. schema issue in an upcoming PR:
* [ ] https://github.com/ssi-dk/diseasystore/pull/121

**Review by SofiaOtero — APPROVED** [review]

**Pull request merged**
